packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -261,8 +258,6 @@ export const ComboBoxDataProviderMixin = (superClass) =>
       if (this.items !== undefined && this.dataProvider !== undefined) {
         restoreOldValueCallback();
         throw new Error('Using `items` and `dataProvider` together is not supported');
-      } else if (this.dataProvider && !this.filteredItems) {
-        this.filteredItems = [];
```

Comment:
**note:** `clearCache()` in `_dataProviderChanged` already ensures the `filteredItems` array is correctly initialized.